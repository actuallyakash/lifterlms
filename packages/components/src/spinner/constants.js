/**
 * Spinner wrapper element class name.
 *
 * @type {string}
 */
export const WRAPPER_CLASSNAME = 'llms-spinning';

/**
 * Spinner element class name.
 *
 * @type {string}
 */
export const CLASSNAME = 'llms-spinner';

/**
 * Small-sized spinner identifier / class name.
 *
 * @type {string}
 */
export const SIZE_SMALL = 'small';

/**
 * Default-sized spinner identifier / class name.
 *
 * @type {string}
 */
export const SIZE_DEFAULT = 'default';
