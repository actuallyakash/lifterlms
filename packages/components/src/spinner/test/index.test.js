import { fakeQuery } from './__MOCKS__/jquery';

import { get, start, stop } from '../';
import { SIZE_SMALL, SIZE_DEFAULT } from '../constants';

global.jQuery = fakeQuery;

describe( 'Spinner: public API', () => {
	describe( 'get()', () => {
		test( 'Autoload styles', () => {
			document.head.innerHTML = '';
			get( 'div', SIZE_DEFAULT, false );
			expect( document.head.innerHTML ).toMatchSnapshot();
		} );

		test( 'Empty return', () => {
			document.body.innerHTML = '<div class="abc"></div>';
			expect( get( '.xyz', SIZE_DEFAULT, false ) ).toBeNull();
		} );

		test( 'Create and return', () => {
			document.body.innerHTML = '<div class="abc"></div>';
			const createResult = get( '.abc', SIZE_DEFAULT, false ),
				body = document.body.innerHTML;
			expect( createResult ).toBeInstanceOf( Element );
			expect( body ).toMatchSnapshot();

			// Don't need to create it again.
			const alreadExistsResult = get( '.abc', SIZE_DEFAULT, false );
			expect( alreadExistsResult ).toBe( createResult );
			expect( document.body.innerHTML ).toBe( body );

			const differentParentResult = get( 'body', SIZE_SMALL, false );
			expect( differentParentResult ).toMatchSnapshot();
			expect( document.body.innerHTML ).toMatchSnapshot();
		} );

		test( 'Create small size', () => {
			document.body.innerHTML = '<div class="abc"></div>';
			const res = get( '.abc', SIZE_SMALL, false ),
				body = document.body.innerHTML;
			expect( res ).toBeInstanceOf( Element );
			expect( body ).toMatchSnapshot();
		} );

		test( 'Return a jQuery selection', () => {
			document.body.innerHTML = '<div class="abc"></div>';
			expect( get( '.abc', SIZE_DEFAULT ) ).toBeInstanceOf( jQuery );
		} );
	} );

	test( 'start() and stop()', () => {
		document.body.innerHTML =
			'<div class="abc xyz"></div><div class="def xyz"></div>';
		get( '.abc' );
		get( '.def' );

		start( '.xyz' );
		expect( document.body.innerHTML ).toMatchSnapshot();

		stop( '.xyz' );
		expect( document.body.innerHTML ).toMatchSnapshot();
	} );
} );
