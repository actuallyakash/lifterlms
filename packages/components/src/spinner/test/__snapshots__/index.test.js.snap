// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner: public API get() Autoload styles 1`] = `"<style id=\\"llms-spinner-styles\\"> .llms-spinning { background: rgba( 250, 250, 250, 0.7 ); bottom: 0; display: none; left: 0; position: absolute; right: 0; top: 0; z-index: 2; } .llms-spinner { animation: llms-spinning 1.5s linear infinite; box-sizing: border-box; border: 4px solid #313131; border-radius: 50%; height: 40px; left: 50%; margin-left: -20px; margin-top: -20px; position: absolute; top: 50%; width: 40px; } .llms-spinner.small { border-width: 2px; height: 20px; margin-left: -10px; margin-top: -10px; width: 20px; } @keyframes llms-spinning { 0% { transform: rotate( 0deg ) } 50% { border-radius: 5%; } 100% { transform: rotate( 220deg) } }</style>"`;

exports[`Spinner: public API get() Create and return 1`] = `"<div class=\\"abc\\"><div class=\\"llms-spinning\\"><i class=\\"llms-spinner default\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div>"`;

exports[`Spinner: public API get() Create and return 2`] = `
<div
  class="llms-spinning"
>
  <i
    aria-live="assertive"
    class="llms-spinner small"
    role="alert"
  >
    <span
      class="screen-reader-text"
    >
      Loading…
    </span>
  </i>
</div>
`;

exports[`Spinner: public API get() Create and return 3`] = `"<div class=\\"abc\\"><div class=\\"llms-spinning\\"><i class=\\"llms-spinner default\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div><div class=\\"llms-spinning\\"><i class=\\"llms-spinner small\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div>"`;

exports[`Spinner: public API get() Create small size 1`] = `"<div class=\\"abc\\"><div class=\\"llms-spinning\\"><i class=\\"llms-spinner small\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div>"`;

exports[`Spinner: public API start() and stop() 1`] = `"<div class=\\"abc xyz\\"><div class=\\"llms-spinning\\" style=\\"display: block;\\"><i class=\\"llms-spinner default\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div><div class=\\"def xyz\\"><div class=\\"llms-spinning\\" style=\\"display: block;\\"><i class=\\"llms-spinner default\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div>"`;

exports[`Spinner: public API start() and stop() 2`] = `"<div class=\\"abc xyz\\"><div class=\\"llms-spinning\\" style=\\"display: none;\\"><i class=\\"llms-spinner default\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div><div class=\\"def xyz\\"><div class=\\"llms-spinning\\" style=\\"display: none;\\"><i class=\\"llms-spinner default\\" role=\\"alert\\" aria-live=\\"assertive\\"><span class=\\"screen-reader-text\\">Loading…</span></i></div></div>"`;
